journal-of-american-mosquito-control-association.csl

```diff
@@ -0,0 +1,261 @@
+          <text term="edition" form="short"/>
+        </group>
+      </if>
+      <else>
+        <text variable="edition" suffix="."/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="collection">
+    <choose>
+      <if type="report">
+        <group prefix=" " suffix="." delimiter=" ">
+          <text variable="collection-title"/>
+          <text variable="number" prefix=" Report No.: "/>
+        </group>
+      </if>
+      <else>
+        <group prefix=" (" suffix=")." delimiter=" ">
+          <names variable="collection-editor" suffix=". ">
+            <name delimiter-precedes-last="always" initialize-with="" name-as-sort-order="all" sort-separator=" "/>
+            <label prefix=", "/>
+          </names>
+          <text variable="collection-title"/>
+        </group>
+      </else>
+    </choose>
+  </macro>
+  <citation et-al-min="3" et-al-use-first="1" disambiguate-add-year-suffix="true" disambiguate-add-names="true" disambiguate-add-givenname="true">
+    <sort>
+      <key macro="year-date"/>
+      <key macro="author-short"/>
+    </sort>
+    <layout prefix="(" suffix=")" delimiter="; ">
+      <group delimiter=" ">
+        <text macro="author-short"/>
+        <text macro="date"/>
+      </group>
+      <text variable="locator" prefix=": "/>
+    </layout>
+  </citation>
+  <bibliography hanging-indent="false" et-al-min="11" et-al-use-first="10">
+    <sort>
+      <key macro="author"/>
+      <key macro="year-date"/>
+    </sort>
+    <layout suffix=".">
+      <group suffix="." delimiter=". ">
+        <text macro="author"/>
+        <text macro="date"/>
+        <text macro="title"/>
+      </group>
+      <group suffix=".">
+        <choose>
+          <if type="bill book graphic legal_case legislation motion_picture report song thesis" match="any">
+            <group prefix=" " suffix="." delimiter=" ">
+              <text macro="edition"/>
+              <text macro="editor"/>
+            </group>
+            <text prefix=" " macro="publisher"/>
+            <text prefix=" " macro="collection"/>
+          </if>
+          <else-if type="chapter paper-conference entry-dictionary entry-encyclopedia" match="any">
+            <group prefix=" " delimiter=" ">
+              <group delimiter=" ">
+                <text term="in" text-case="capitalize-first" suffix=":"/>
+                <text macro="editor"/>
+                <text variable="container-title" suffix="."/>
+              </group>
+              <text variable="volume" prefix="Vol. " suffix="."/>
+              <text macro="edition"/>
+              <group suffix="." delimiter=". ">
+                <text macro="publisher"/>
+                <text prefix=" " macro="collection"/>
+                <text macro="pages"/>
+              </group>
+            </group>
+          </else-if>
+          <else-if type="review review-book" match="any">
+            <text macro="editor" prefix=" " suffix="."/>
+            <group prefix=" ">
```

Comment:
Throughout the style, you frequently use prefixes and suffixes to hold spaces and other delimiters between elements. This is a bad practice, as it can lead to extraneous spaces or punctuation when a variable is missing. For example, if a journal article didn't have a volume, issue, or page number, the style currently would add 2 `.` between the container title and the DOI. 

Please go through your style and consistently use the `delimiter` argument on `<group>` to hold delimiters between elements, rather than prefix/suffix. Only use prefix/suffix for something that should always appear on the element, regardless of what other variables are or aren't there (e.g., the `()` around issue).